### PR TITLE
Add CIS Microsoft 365 Foundations Benchmark v2.0.0 - (03-31-2023). Closes #36

### LINKED ISSUE #36: Add CIS Microsoft 365 Foundations Benchmark v2.0.0 - (03-31-2023)
**Is your feature request related to a problem? Please describe.**

CIS [Reference ](https://www.cisecurity.org/benchmark/microsoft_365)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by madhushreeray30**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: cis_v200/section_1.sp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    cis_type              = "manual"
+    microsoft_365_license = "E3"
+    service               = "Azure/ActiveDirectory"
+  })
+}
+
+control "cis_v200_1_1_4" {
+  title         = "1.1.4 Ensure multifactor authentication is enabled for all users"
+  description   = "Enable multifactor authentication for all users in the Microsoft 365 tenant. Users will be prompted to authenticate with a second factor upon logging in to Microsoft 365 services. The second factor is most commonly a text message to a registered mobile phone number where they type in an authorization code, or with a mobile application like Microsoft Authenticator."
+  query         = query.azuread_admin_user_mfa_enabled
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:100]:
```suggestion
  query         = query.azuread_all_user_mfa_enabled
```

File: cis_v200/section_1.sp [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    cis_type              = "manual"
+    microsoft_365_license = "E3"
+    service               = "Azure/ActiveDirectory"
+  })
+}
+
+control "cis_v200_1_1_6" {
+  title         = "1.1.6 Ensure 'Phishing-resistant MFA strength' is required for Administrators"
+  description   = "Authentication strength is a Conditional Access control that allows administrators to specify which combination of authentication methods can be used to access a resource. For example, they can make only phishing-resistant authentication methods available to access a sensitive resource. But to access a non-sensitive resource, they can allow less secure multifactor authentication (MFA) combinations, such as password + SMS. Microsoft has 3 built-in authentication strengths. MFA strength, Passwordless MFA strength, and Phishing-resistant MFA strength. Ensure administrator roles are using a CA policy with Phishing-resistant MFA strength."
+  query         = query.azuread_legacy_authentication_disabled
```
Comment (khushboo9024) [inline cis_v200/section_1.sp:130]: Is this query reference correct?

File: cis_v200/section_1.sp [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  ]
+
+  tags = merge(local.cis_v200_1_1_common_tags, {
+    type    = "Benchmark"
+    service = "Azure/ActiveDirectory"
+  })
+}
+
+control "cis_v200_1_1_1" {
+  title         = "1.1.1 Ensure Security Defaults is disabled on Azure Active Directory"
```
Comment (misraved) [inline cis_v200/section_1.sp:51]:
```suggestion
  title         = "1.1.1 Ensure Security Defaults are disabled on Azure Active Directory"
```

File: cis_v200/section_2.sp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    cis_level             = "1"
+    cis_type              = "automated"
+    microsoft_365_license = "E3"
+    service               = "Azure/ActiveDirectory"
+  })
+}
+
+control "cis_v200_2_2" {
+  title         = "2.2 Ensure third party integrated applications are not allowed"
+  description   = "App registrations allows users to register custom-developed applications for use within the directory."
```
Comment (misraved) [inline cis_v200/section_2.sp:40]:
```suggestion
  description   = "App registrations allow users to register custom-developed applications for use within the directory."
```

**Review by misraved — COMMENTED** [review]
@khushboo9024 please take a look at the review comments.

Also, could you please update the doc files to include v2.0.0 references?

**Review by misraved — APPROVED** [review]

**Pull request merged**